controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -332,27 +359,27 @@ func (r *OCSInitializationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 					Name:      util.OcsOperatorConfigName,
 					Namespace: r.OperatorNamespace,
 				},
-			},
-			handler.EnqueueRequestsFromMapFunc(
-				func(context context.Context, obj client.Object) []reconcile.Request {
-					return []reconcile.Request{{
-						NamespacedName: InitNamespacedName(),
-					}}
-				},
-			),
+			}, enqueueOCSInit,
 		).
 		// Watcher for prometheus operator csv
 		Watches(
 			&opv1a1.ClusterServiceVersion{},
-			handler.EnqueueRequestsFromMapFunc(
-				func(context context.Context, obj client.Object) []reconcile.Request {
-					return []reconcile.Request{{
-						NamespacedName: InitNamespacedName(),
-					}}
-				},
-			),
+			enqueueOCSInit,
 			builder.WithPredicates(prometheusPredicate),
+		).
+		Watches(&extv1.CustomResourceDefinition{}, enqueueOCSInit, builder.WithPredicates(crdPredicate), builder.OnlyMetadata)
+
+	if r.AvailableCrds[ClusterClaimCrdName] {
+		ocsInitializationController = ocsInitializationController.Watches(
+			&v1alpha1.ClusterClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: util.OdfInfoNamespacedNameClaimName,
+				},
+			},
```

Comment:
This will not do what you expect it to do. This will watch all `ClusterClaims` not only the one with the name specified by `util.OdfInfoNamespacedNameClaimName`. The code uses only the GVK of the passed object to establish the watch. If you want to filter based on name you will need a predicate and you can use the one I suggested adding in the last comment (filter base on resource name)